src/utils/convert.js

```diff
@@ -825,278 +194,257 @@ const renderDefaultInputData = definition => {
+  let exportBlock = _.cloneDeep(appDefinition.authentication),
+    functionBlock = '';
+
+  _.each(
+    {
+      connectionLabel: 'getConnectionLabel',
+      test: 'testAuth'
+    },
+    (funcName, key) => {
+      const func = appDefinition.authentication[key];
+      if (func && func.source) {
+        const args = func.args || ['z', 'bundle'];
+        functionBlock += `const ${funcName} = (${args.join(', ')}) => {${
+          func.source
+        }};\n\n`;
+
+        exportBlock[key] = makePlaceholder(funcName);
+      }
+    }
   );
-};
 
-const renderUtils = () => {
-  const templateFile = path.join(TEMPLATE_DIR, '/utils.template.js');
-  return renderTemplate(templateFile);
-};
+  exportBlock = `module.exports = ${replacePlaceholders(
+    JSON.stringify(exportBlock)
+  )};\n`;
 
-const writeUtils = newAppDir => {
-  const fileName = 'utils.js';
-  return renderUtils().then(content =>
-    createFile(content, fileName, newAppDir)
-  );
+  return prettifyJs(functionBlock + exportBlock);
 };
 
-const findSearchOrCreates = legacyApp => {
-  let searchOrCreates = {};
-  _.each(legacyApp.searches, (searchDef, searchKey) => {
-    if (searchDef.action_pair_key) {
-      // The key for a searchOrCreate (comboKey) has to match the key of a
-      // search due to frontend constraints. From Platform's perspective
-      // though, a searchOrCreate just needs a unique key that could be
-      // anything.
-      const comboKey = searchKey;
-      searchOrCreates[comboKey] = {
-        key: comboKey,
-        display: {
-          label: searchDef.action_pair_label || '',
-          description: searchDef.action_pair_label || ''
-        },
-        search: searchKey,
-        create: searchDef.action_pair_key
-      };
+const renderHydrators = async appDefinition => {
+  let exportBlock = _.cloneDeep(appDefinition.hydrators),
+    functionBlock = '';
+
+  _.each(appDefinition.hydrators, (func, funcName) => {
+    if (func && func.source) {
+      const args = func.args || ['z', 'bundle'];
+      functionBlock += `const ${funcName} = (${args.join(', ')}) => {${
+        func.source
+      }};\n\n`;
+      exportBlock[funcName] = makePlaceholder(funcName);
     }
   });
-  return searchOrCreates;
-};
 
-const renderIndex = legacyApp => {
-  const needsAuth = hasAuth(legacyApp);
-  const templateContext = {
-    HEADER: '',
-    TRIGGERS: '',
-    SEARCHES: '',
-    CREATES: '',
-    BEFORE_REQUESTS: getBeforeRequests(legacyApp),
-    AFTER_RESPONSES: getAfterResponses(legacyApp),
-    needsAuth
-  };
+  exportBlock = `module.exports = ${replacePlaceholders(
+    JSON.stringify(exportBlock)
+  )};\n`;
 
-  return getHeader(legacyApp).then(header => {
-    templateContext.HEADER = header;
+  return prettifyJs(functionBlock + exportBlock);
+};
 
-    const importLines = [];
+const renderIndex = async appDefinition => {
+  let exportBlock = _.cloneDeep(appDefinition),
+    functionBlock = '',
+    importBlock = '';
```

Comment:
I'd probably do importBlock as an array and then `.join('\n')` at the end instead of needing to use a newline whenever you add to it.

Reply:
Good call! Changed in 3447bd7.